addons/mail/static/src/models/thread/thread.js

```diff
@@ -1464,7 +1464,13 @@ function factory(dependencies) {
         * @returns {boolean}
         */
         _computeIsChannelDescriptionChangeable() {
-            return this.model === 'mail.channel' && ['channel', 'group'].includes(this.channel_type);
+            return (
+                this.model === 'mail.channel' &&
+                ['channel', 'group'].includes(this.channel_type) &&
+                this.messaging.currentUser &&
+                !this.messaging.currentUser.isPortalUser &&
+                !this.messaging.isCurrentUserGuest
```

Comment:
Looks like the last line is not needed, if `currentUser` is set I think `isCurrentUserGuest` is necessarily false.